### DISCUSSION TITLE
How hard would it be to add support for custom and exotic compilers?

### DISCUSSION THREAD

**Discussion by nanoni17728** [microsoft/vscode-cpptools#7402] [Ideas]
Hi!

I’m currently learning to build games for the [Pokemon Mini](https://www.pokemon-mini.net/). The tooling around the [CPU](http://www.rayslogic.com/Software/TimexUSB/Docs/s1c88%20core%20cpu%20manual.pdf) was made in-house by Epson as far as I know and is proprietary. As such, when I try to specify the path to the compiler with `compilerPath` in `c_cpp_properties.json` the extension tells me `Unable to resolve configuration with compilerPath [...]`. From my understanding, it seems to be the expected behavior, since people reported the same kind of issue with other less mainstream tools, like [there](https://github.com/microsoft/vscode-cpptools/issues/7146).

With such a niche CPU and tooling, it is definitely not worth for you as maintainers to add support for the extension. However, I was wondering how hard it would be for someone else (me in that case) to add support for niche compilers? I’m quite interested to do that kind of work.

If you need me to give you more inputs, ask away. Also if you’re aware of simpler workarounds I would be glad to be told about it :)

Thanks for your time,

**Comment by sean-mcmanus** [comment]
We recently worked with NVIDIA to add IntelliSense support for their CUDA extension. You or the compiler developers would need to get the https://www.edg.com/ compiler (which is a paid license) and then add/push changes that would makes it easy for us to turn on a mode that enables the language extensions required. I think it would be hard to do.

As a workaround you could try setting compilerPath to "" and then manually set any required includePath/defines.

**Reply by nanoni17728** [reply]
Thank you for your answer! If I understood correctly, if people want proper support of their compiler with IntelliSense, they have to have access to the sources, right? The current compiler we use is proprietary and we only have access to its executable.

I’ll set the compilerPath to an empty string for the moment.

**Reply by sean-mcmanus** [reply]
You wouldn't need access to the proprietary compiler's source code, but you'd need to understand it's behavior enough in regards to all the non-standard C/C++ to make any changes necessary to the EDG front-end compiler to mimic its behavior, e.g. new keywords, etc.

**Reply by nanoni17728** [reply]
Alright. Thank you for your time
